--- CMakeLists.txt	2020-06-23 19:45:38.802651400 +0200
+++ CMakeLists.txt	2020-06-23 19:47:44.341851200 +0200
@@ -5,4 +5,4 @@
 cegui_add_dependency(${CEGUI_TARGET_NAME} SDL2 PRIVATE TRUE)
 cegui_add_dependency(${CEGUI_TARGET_NAME} SDL2IMAGE PRIVATE TRUE)
 
-cegui_target_link_libraries(${CEGUI_TARGET_NAME} ${CEGUI_BASE_LIBNAME})
+cegui_target_link_libraries(${CEGUI_TARGET_NAME} ${CEGUI_BASE_LIBNAME} ${CONAN_LIBS})
